src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -1082,6 +1086,30 @@ public unsafe void CopyFromAscii(string data)
             _index = blockIndex;
         }
 
+        private static byte[] NumericBytesScratch => _numericBytesScratch ?? CreateNumericBytesScratch();
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static byte[] CreateNumericBytesScratch()
+        {
+            return (_numericBytesScratch = new byte[_maxPositiveLongByteLength]);
+        }
+
+        public void CopyFromNumeric(long value)
```

Comment:
Changing to `ulong` as it doesn't handle negatives; either that or make internal